### PR TITLE
Add new fields in VMware and BareMetal cluster resources

### PR DESCRIPTION
**Pull request by naitianliu-google**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
gkeonprem: added `upgrade_policy` field in `google_gkeonprem_vmware_cluster` resource (beta)
```

```release-note:enhancement
gkeonprem: added `upgrade_policy` and `binary_authorization` fields in `google_gkeonprem_bare_metal_cluster` resource (beta)
```

### CODE REVIEW

**Comment by zli82016** [comment]
Thanks for making the changes. Can you please add an update test for the new fields? 
Docs to add the update tests: https://googlecloudplatform.github.io/magic-modules/develop/test/#add-an-update-test

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/products/gkeonprem/BareMetalCluster.yaml [hunk 1]
```diff
@@ -747,6 +747,30 @@ properties:
...
+    name: 'binaryAuthorization'
+    description: Binary Authorization related configurations.
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'evaluationMode'
+        description: |
+          Mode of operation for binauthz policy evaluation. If unspecified, 
+          defaults to DISABLED.
+        values:
+          - EVALUATION_MODE_UNSPECIFIED
```
[the remark this answers is not in the document]
Comment (naitianliu-google) [reply mmv1/products/gkeonprem/BareMetalCluster.yaml]: Removed. Thanks for the guidance!

**Comment by naitianliu-google** [comment]
> Thanks for making the changes. Can you please add update tests for the new fields? Docs to add the update tests: https://googlecloudplatform.github.io/magic-modules/develop/test/#add-an-update-test

Thanks for pointing this out. Added the new fields in update test. 

Please help take a pass and let me know if anything else needs to be updated. Thanks! @zli82016

File: mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb [hunk 2]
```diff
@@ -381,6 +393,12 @@ func testAccGkeonpremBareMetalCluster_bareMetalClusterUpdateManualLb(context map
         }
       }
     }
+    binary_authorization {
+      evaluation_mode = "DISABLED"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb:397]: Can you please update this mode to `PROJECT_SINGLETON_POLICY_ENFORCE` to test the update of this field from `DISABLED`?

File: mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb [hunk 3]
```diff
@@ -381,6 +393,12 @@ func testAccGkeonpremBareMetalCluster_bareMetalClusterUpdateManualLb(context map
...
+    }
+    upgrade_policy {
+      policy = "SERIAL"
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb:400]: Can you please change the value to `CONCURRENT` to test the update of this field from `SERIAL`?

**Comment by zli82016** [comment]
Sorry, I was OOO last week. Can you please address the new comments? Thanks.

File: mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb (hunk 2 above)
Comment (naitianliu-google) [reply mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb]: Updated. Thanks for the suggestion!

File: mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb (hunk 3 above)
Comment (naitianliu-google) [reply mmv1/third_party/terraform/services/gkeonprem/resource_gkeonprem_bare_metal_cluster_test.go.erb]: Updated. Thanks for checking!

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks for making the changes.

**Pull request merged**